Qveo-Crud-Usuario

M4isth3r opened the pull request:
- Guardar usuario
 - Mostrar lista de usuarios
 - eliminar usuario con ajax
 - mostrar usuario con ajax
 - he agregado el chipPlataformas en un jsp aparte
 - falta por mejorar las validaciones en algunos campos
 - me daba problemas cuando implementaba el confirmar con model...

M4isth3r committed:
Qveo-Crud-Usuario

 - Guardar usuario
 - Mostrar lista de usuarios
 - eliminar usuario con ajax
 - mostrar usuario con ajax
 - he agregado el chipPlataformas en un jsp aparte
 - falta por mejorar las validaciones en algunos campos
 - me daba problemas cuando implementaba el confirmar con model...

M4isth3r committed:
Modifcadas validaciones con interfaz validator

- Falta por validar campos como sexo, pais, email y password

- No borra imagen si ya existe...

JaviDareku commented on src/main/java/com/qveo/qveoweb/controller/UsuarioController.java:
```diff
@@ -0,0 +1,134 @@
+@Controller
+public class UsuarioController {
+
+	private boolean editar = false;
```
Es una mala práctica en las aplicaciones web el uso de variables de clase. Si dos usuarios a la vez editan o dan de alta usuarios comparten esta variable y puede ocasionar errores

M4isth3r replied:
Lo utilizan para luego el tema de file... hay que darle otra vuelta Y me tienes que explicar lo de los errores XD

JaviDareku commented on src/main/java/com/qveo/qveoweb/controller/UsuarioController.java:
```diff
@@ -0,0 +1,134 @@
+			
+			if (!file.isEmpty()) {
+				if(editar) {
+					
```
Esta lógica es susceptible de sacarse a un servicio. Lo ideal es que los controladores no tengan muchas logicas

M4isth3r replied:
La verdad es que la parte de file yo no la he hecho, así que no la he revisado mucho, pero le daré una vuelta. Además recuerdo que nos digite que toda la lógica la intentaramos dejar en los servicios

JaviDareku commented on src/main/java/com/qveo/qveoweb/controller/UsuarioController.java:
```diff
@@ -0,0 +1,134 @@
+				usuario.setFoto("/resources/img/usuarios/"+uniqueFilename);
+			}
+			
+			if(file.isEmpty() && editar) {
```
Si se quiere editar un usuario no es necesario volver a setear la foto que tenía porque teóricamente no se ha editado

JaviDareku commented on src/main/java/com/qveo/qveoweb/service/Imp/PaisServiceImp.java:
```diff
@@ -0,0 +1,24 @@
+
+	@Override
+	public List<Pais> getAllPais() {
+		// TODO Auto-generated method stub
```
Estos comentarios borradlos jajajaja

JaviDareku commented on src/main/java/com/qveo/qveoweb/service/Imp/UploadFileServiceImpl.java:
```diff
@@ -0,0 +1,106 @@
+		
+		Path rootPath = getPath(nombreFinal,accion);
+
+		log.info("rootPath: " + rootPath);
```
Los logs es recomendable currarselos más. Que aporten información relevante

JaviDareku commented on src/main/java/com/qveo/qveoweb/service/Imp/UploadFileServiceImpl.java:
```diff
@@ -0,0 +1,106 @@
+	public Path getPath(String filename,Integer accion) {
+		String ruta = "";
+		switch (accion) {
+		case 1:
```
Buena idea, como recomendación para que el código sea más legible es mejor usar un Enum

JaviDareku commented on src/main/java/com/qveo/qveoweb/service/Imp/UsuarioServiceImp.java:
```diff
@@ -0,0 +1,67 @@
+
+	@Override
+	public List<Usuario> findUsuarioPorNombre(String nombre) {
+		
```
Es recomendable no poner nombres en los servicios que puedan parecer de JPA

JaviDareku commented on src/main/webapp/resources/js/usuario/lista.js:
```diff
@@ -0,0 +1,171 @@
+		type : 'GET',
+		url : '/qveo/ajax/usuario/' + id,
+		success : function(data) {
+
```
Javascript es feo pero también se puede modular. Intentad evitar estos métodos tan largos

JaviDareku commented on src/main/java/com/qveo/qveoweb/controller/UsuarioController.java:
```diff
@@ -0,0 +1,140 @@
+				usuario.setFoto("/resources/img/usuarios/"+uniqueFilename);
+			}
+			
+			if(file.isEmpty()) {
```
No se si se ha perdido el comentario que puse aquí jajajaja Si estas editando y no viene el fichero del formulario no veo la necesidad de volver a setearle la antigua foto. Teoricamente el objeto la tiene ya informada

AchrafAdev replied:
No se porque pero la setea null, de alli que hayamos realizado ese setter

M4isth3r committed:
Modificaciones de Mostrar usuario en listar

- El código de mostrar lo he subdividido en varias funciones
- También he arreglado el error de eliminar usuario con confirmación con modal de materialize
- He agregado las validaciones, pero no me funcionan pais, sexo, plataformas

AchrafAdev committed:
Cambios Validacion

M4isth3r committed:
Auto stash before merge of "Qveo_Crud_Usuario" and "origin/Qveo_Crud_Usuario"

M4isth3r committed:
Modificado el fileUpload 

- Agregado foto por defecto
